ember-client/shared/src/main/scala/org/http4s/ember/client/EmberClient.scala

```diff
@@ -44,7 +44,12 @@ final class EmberClient[F[_]] private[client] (
   ): F[Throwable] = F match {
     case concurrentF: Concurrent[F @unchecked] =>
       implicit val C = concurrentF
-      resp.body.compile.drain.as(UnexpectedStatus(resp.status, req.method, req.uri))
+      resp.entity match {
+        case Entity.Empty | Entity.Strict(_) =>
+          F.pure(UnexpectedStatus(resp.status, req.method, req.uri))
+        case Entity.Default(body, _) =>
+          body.compile.drain.as(UnexpectedStatus(resp.status, req.method, req.uri))
```

Comment:
The readability here will be super nice when/if https://github.com/http4s/http4s/pull/6305 merges :sunglasses: